### PR TITLE
Tests for token balance checks

### PR DESCRIPTION
**Pull request by uzyn**

### CODE REVIEW

File: src/amount.h [hunk 1]
```diff
@@ -100,7 +100,12 @@ struct CTokenAmount { // simple std::pair is less informative
     CAmount nValue;
 
     std::string ToString() const {
-        return std::to_string(nValue / COIN) + "." + std::to_string(nValue % COIN) + "@" + nTokenId.ToString();
+        // copy of ValueFromAmount()
+        bool sign = nValue < 0;
+        int64_t n_abs = (sign ? -nValue : nValue);
+        int64_t quotient = n_abs / COIN;
+        int64_t remainder = n_abs % COIN;
```
Comment (ShengguangXiao) [inline src/amount.h:107]: add const for these variables

File: src/test/applytx_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    return scriptMeta;
+}
+
+// redundant due to 'neg_token_amounts'
+BOOST_AUTO_TEST_CASE(apply_a2a_neg)
+{
+    Consensus::Params amkCheated = Params().GetConsensus();
+    amkCheated.AMKHeight = 0;
+
+
```
Comment (ShengguangXiao) [inline src/test/applytx_tests.cpp:83]: Remove extra blank line

File: src/test/applytx_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    BOOST_CHECK_EQUAL(mnview.GetBalance(owner, DFI), dfi100);
+
+    // create templates for msg and tx:
+    CAccountToAccountMessage msg{};
+    msg.from = owner;
+    CMutableTransaction rawTx;
+    rawTx.vout = { CTxOut(0, CScript()) };
+    rawTx.vin = { CTxIn(auth_out) };
+
+    // try to send "A:-1@DFI"
```
Comment (ShengguangXiao) [inline src/test/applytx_tests.cpp:107]: Can you also add fail reason in the comment?

File: src/test/applytx_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,176 @@
+#include <chainparams.h>
```
Comment (ShengguangXiao) [inline src/test/applytx_tests.cpp:1]: @uzyn @monstrobishi Do we need add defichain copyright information for the new files added by us?

**Comment by monstrobishi** [comment]
@DeFiDev will you look at some of @ShengguangXiao suggestions and also update the branch

**Pull request merged**